### ISSUE TITLE
LPC1768: dtostrf function not working, (arduino floating point not working)

### ISSUE THREAD

**Issue by robbycandra** [MarlinFirmware/Marlin#15883]
**Description**

I use LPC1768
dtostrf function not working, because sprintf is not working for floating point.

**Steps to Reproduce**

Try M114 D with M114_DETAIL enabled
It also affect PowerLoss Recovery

**Comment by thinkyhead** [comment]
Strange. Arduino recommends using `dtostrf` because `%f` is not supported by the `printf` functions.

**Comment by thinkyhead** [comment]
This is what the LPC platform's `arduino.cpp` does for `dtostrf`…
```cpp
char *dtostrf (double __val, signed char __width, unsigned char __prec, char *__s) {
  char format_string[20];
  snprintf(format_string, 20, "%%%d.%df", __width, __prec);
  sprintf(__s, format_string, __val);
  return __s;
}
```
…and sure enough, it is trying to use `%f`. Looks like an LPC platform bug.

**Comment by thinkyhead** [comment]
Looking at other platforms, I see that most of them implement `dtostrf` in just the same way. Considering that `dtostrf` has been working, and this includes the LPC1768 machines I am testing, then perhaps some breaking change has recently been made to the LPC platform…?

**Comment by p3p** [comment]
It was my fault, I disabled printf float support, thinking that I was using dtostrf from the linked `m` library, though that appears to be AVR specific.

**Comment by thinkyhead** [comment]
> I disabled `printf` `float` support

Where is that disabled?

**Comment by p3p** [comment]
In the platform, just patched it in release 0.1.2, will update platformio.ini

**Comment by robbycandra** [comment]
Problem Fixed; Thanks
